src/lib/optimizer/optimizer.cpp

```diff
@@ -84,24 +84,27 @@ namespace opossum {
 std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   auto optimizer = std::make_shared<Optimizer>(100);
 
-  RuleBatch initial_batch(RuleBatchExecutionPolicy::Once);
-  initial_batch.add_rule(std::make_shared<ConstantCalculationRule>());
+  RuleBatch final_batch(RuleBatchExecutionPolicy::Once);
+  final_batch.add_rule(std::make_shared<ConstantCalculationRule>());
 
   // Run pruning just once since the rule would otherwise insert the pruning ProjectionNodes multiple times.
-  initial_batch.add_rule(std::make_shared<ColumnPruningRule>());
-
-  optimizer->add_rule_batch(initial_batch);
+  final_batch.add_rule(std::make_shared<ColumnPruningRule>());
 
-  RuleBatch main_batch(RuleBatchExecutionPolicy::Iterative);
-  main_batch.add_rule(std::make_shared<PredicatePushdownRule>());
-  main_batch.add_rule(std::make_shared<ExistsReformulationRule>());
-  optimizer->add_rule_batch(main_batch);
+  final_batch.add_rule(std::make_shared<ExistsReformulationRule>());
```

Comment:
Just out of curiosity, why in the final batch? I remember a paper that argued that join reformulations should happen early as many other simple rules might rewrite queries that then no longer can be rewritten to joins.

Reply:
As of this PR, there is only one, non-iterative batch left. Could also call it the "initial_batch", it doesn't matter.